src/main/java/core/basesyntax/OperationStrategyImpl.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> map;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> map) {
+        this.map = map;
+    }
+
+    @Override
+    public void getHandler(FruitTransaction operation) {
+        map.get(operation.getActivity()).run(operation);
+    }
```

Comment:
Aim of strategy is to provide proper implementation. It shouldn't call any methods from chosen one

```suggestion
    @Override
    public OperationHandler getHandler(FruitTransaction operation) {
        map.get(operation.getActivity());
    }
```

Reply:
Changed but I'm not sure